src/commands/internaltesting-functions-discover.ts

```diff
@@ -0,0 +1,22 @@
+import { Command } from "../command";
+import { Options } from "../options";
+import { logger } from "../logger";
+import { loadCodebases } from "../deploy/functions/prepare";
+import { normalizeAndValidate } from "../functions/projectConfig";
+import { getProjectAdminSdkConfigOrCached } from "../emulator/adminSdkConfig";
+import { needProjectId } from "../projectUtils";
+
+import type { FirebaseConfig } from "../deploy/functions/args";
+
+export const command = new Command("internaltesting:functions:discover")
+  .description("Discover function triggers defined in the current project directory")
+  .action(async (options: Options) => {
+    const projectId = needProjectId(options);
+    const fnConfig = normalizeAndValidate(options.config.src.functions);
+    const firebaseConfig = (await getProjectAdminSdkConfigOrCached(projectId)) as FirebaseConfig;
```

Comment:
can this `as` be removed? (maybe `getProjectAdminSdkConfigOrCached` typed? if not, maybe add a TODO to clean up the `as`?